08_cell_line_prediction/nbconverted/download_data.py

```diff
@@ -0,0 +1,353 @@
+# 
+# This should help us with setting sensible thresholds for building classifiers, since the dataset overall is considerably smaller than TCGA.
+
+# In[14]:
+
+
+# gene to explore mutation distribution for
+gene = 'RB1'
+
+gene_index = (ccle_sample_info_df.index
+    .intersection(ccle_expression_df.index)
+    .intersection(ccle_mutation_binary_df[ccle_mutation_binary_df[gene] == 1].index)
+)
+print(gene_index.shape)
+
+
+# In[15]:
+
+
+ccle_gene_cancer_types = (ccle_sample_info_df
+    .reindex(gene_index)
+    .groupby('primary_disease')
+    .count()
+    .reset_index()
+    .iloc[:, [0, 1]]
+    .rename(columns={'cell_line_name': '{}_count'.format(gene)})
+    .merge(ccle_cancer_types, how='right', on='primary_disease')
+    .fillna(value=0)
+)
+
+ccle_gene_cancer_types['{}_proportion'.format(gene)] = (
+    ccle_gene_cancer_types['{}_count'.format(gene)] / ccle_gene_cancer_types['count']
+)
+
+ccle_gene_cancer_types.head()
+
+
+# In[16]:
+
+
+ccle_gene_tissues = (ccle_sample_info_df
+    .reindex(gene_index)
+    .groupby('lineage')
+    .count()
+    .reset_index()
+    .iloc[:, [0, 1]]
+    .rename(columns={'cell_line_name': '{}_count'.format(gene)})
+    .merge(ccle_tissues, how='right', on='lineage')
+    .fillna(value=0)
+)
+ccle_gene_tissues['{}_proportion'.format(gene)] = (
+    ccle_gene_tissues['{}_count'.format(gene)] / ccle_gene_tissues['count']
+)
+
+ccle_gene_tissues.head()
+
+
+# In[17]:
+
+
+sns.set({'figure.figsize': (18, 10)})
+fig, axarr = plt.subplots(2, 1)
+
+sns.barplot(data=ccle_gene_cancer_types, x='primary_disease',
+            y='{}_count'.format(gene),
+            ax=axarr[0])
+axarr[0].set_xticklabels(axarr[0].get_xticklabels(), rotation=90)
+axarr[0].set_xlabel('Cancer type')
+axarr[0].set_ylabel('Count')
+
+sns.barplot(data=ccle_gene_tissues, x='lineage',
+            y='{}_count'.format(gene),
+            ax=axarr[1])
+axarr[1].set_xticklabels(axarr[1].get_xticklabels(), rotation=90)
+axarr[1].set_xlabel('Tissue lineage')
+axarr[1].set_ylabel('Count')
+
+plt.suptitle(
+    'Distribution of cell lines with {} mutation across cancer types/tissues'.format(gene)
+)
+plt.tight_layout()
+
+
+# In[18]:
+
+
+sns.set({'figure.figsize': (18, 10)})
+fig, axarr = plt.subplots(2, 1)
+
+sns.barplot(data=ccle_gene_cancer_types, x='primary_disease',
+            y='{}_proportion'.format(gene),
+            ax=axarr[0])
+axarr[0].set_xticklabels(axarr[0].get_xticklabels(), rotation=90)
+axarr[0].set_xlabel('Cancer type')
+axarr[0].set_ylabel('Proportion')
+axarr[0].set_ylim(0.0, 1.0)
+
+sns.barplot(data=ccle_gene_tissues, x='lineage',
+            y='{}_proportion'.format(gene),
+            ax=axarr[1])
+axarr[1].set_xticklabels(axarr[1].get_xticklabels(), rotation=90)
+axarr[1].set_xlabel('Tissue lineage')
+axarr[1].set_ylabel('Proportion')
+axarr[1].set_ylim(0.0, 1.0)
+
+plt.suptitle(
+    'Proportion of cell lines with {} mutation across cancer types/tissues'.format(gene)
```

Comment:
You only have RB1 in the notebook but the code looks designed to screen multiple mutations. Are you just starting with RB1 and plan to try other genes later? Or did you try multiple and just finish with RB1?

Reply:
Good idea - I tried multiple and just finished with RB1, but maybe it would be best to run a few examples using Paperpile [like I did here](https://github.com/greenelab/pancancer-evaluation/blob/master/02_cancer_type_classification/scripts/run_fs_for_genes.sh).